extensions/podman/packages/extension/src/extension.ts

```diff
@@ -2010,6 +2018,16 @@ export async function createMachine(
     parameters.push('--image-path');
     parameters.push(params['podman.factory.machine.image-path']);
     telemetryRecords.imagePath = 'custom';
+  } else if (params['podman.factory.machine.image-uri']) {
+    const imageUri = params['podman.factory.machine.image-uri'].trim();
+    parameters.push('--image-path');
+    if (imageUri.startsWith('https://') || imageUri.startsWith('http://')) {
+      parameters.push(imageUri);
+      telemetryRecords.imagePath = 'custom-url';
+    } else {
+      parameters.push(`docker://${imageUri}`);
+      telemetryRecords.imagePath = 'custom-registry';
```

Comment:
should we send the hash of the URL/images ?

Reply:
I opened #9178 to discuss what we can send in telemetry when custom image is used.